src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/hooks.ts

```diff
@@ -0,0 +1,114 @@
+import {useFirestoreConfiguration} from '@atb/configuration';
+import {Location} from '@atb/favorites';
+import {CityZone} from '@atb/reference-data/types';
+import {onlyUniquesBasedOnField} from '@atb/utils/only-uniques';
+import turfBooleanPointInPolygon from '@turf/boolean-point-in-polygon';
+import {useMemo} from 'react';
+import {
+  Modes,
+  StreetMode,
+} from '@atb/api/types/generated/journey_planner_v3_types';
+import {useFlexibleTransportEnabled} from './use-flexible-transport-enabled';
+import {StorageModelKeysEnum} from '@atb/storage';
+import {useDebugOverride} from '@atb/debug';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+import {RemoteConfigKeys} from '@atb/remote-config';
+
+export const useFindCityZoneInLocation = (
+  location: Location | undefined,
+  cityZones?: CityZone[],
+) => {
+  return useMemo(() => {
+    if (location && cityZones) {
+      const {longitude, latitude} = location.coordinates;
+      return cityZones.find(({geometry}) =>
+        turfBooleanPointInPolygon([longitude, latitude], geometry),
+      );
+    }
+  }, [
+    cityZones,
+    location?.coordinates.longitude,
+    location?.coordinates.latitude,
+  ]);
+};
+
+export const useFindCityZonesInLocations = (
+  from: Location | undefined,
+  to: Location | undefined,
+): CityZone[] | undefined => {
+  const {cityZones} = useFirestoreConfiguration();
+  const fromCityZone = useFindCityZoneInLocation(from, cityZones);
+  const toCityZone = useFindCityZoneInLocation(to, cityZones);
+
+  const filteredCityZones: CityZone[] = [];
+  fromCityZone &&
+    fromCityZone.enabled == true &&
+    filteredCityZones.push(fromCityZone);
+  toCityZone &&
+    toCityZone.enabled === true &&
+    filteredCityZones.push(toCityZone);
+
+  return filteredCityZones.filter(onlyUniquesBasedOnField('name'));
```

Comment:
This works without casting. Another alternative similar to the old implementation is to use type guard (`cityZone is CityZone`) instead of casting. I'll just give some examples, but feel free to use the implementation that you think is best.

So instead of casting:
```ts
const filteredCityZones = [fromCityZone, toCityZone].filter(
  (cityZone) => cityZone && cityZone.enabled === true,
) as CityZone[]
```

it would be with a type guard: 
```ts
const filteredCityZones = [fromCityZone, toCityZone].filter(
  (cityZone): cityZone is CityZone => !!cityZone && cityZone.enabled === true,
)
```

Also possible here is an even shorter version:
```ts
return [fromCityZone, toCityZone]
  .filter((cityZone): cityZone is CityZone => !!cityZone?.enabled)
  .filter(onlyUniquesBasedOnField('name'));
```

Reply:
Great suggestion, I have implemented the shortest version.